draft-irtf-cfrg-hpke.md

```diff
@@ -1302,8 +1349,15 @@ Template:
 * Value: The two-byte identifier for the algorithm
 * KEM: The name of the algorithm
 * Nzz: The length in bytes of a KEM shared secret produced by the algorithm
-* Nenc: The length in bytes of an encapsulated key produced by the algorithm
-* Npk: The length in bytes of an encoded public key for the algorithm
+* Nenc: The length in bytes of an encapsulated key produced by the algorithm.
```

Comment:
Maybe this should be the maximum length of an encapsulated key? The text below (suggesting that there are different lengths that need to be specified) seems odd. Really this value is in place to help with allocations and whatnot, right?

Reply:
I like your idea of the maximum length. For the NIST P-curves, the deserialize algorithm is actually specified this way: it takes up to the maximum length, and a condition inside decides if decompression is done, depending on the actually received size.